nbstripout/_utils.py

```diff
@@ -153,3 +157,106 @@ def strip_output(nb, keep_output, keep_count, extra_keys=[], drop_empty_cells=Fa
+
+    return dict_config
+
+
+def process_setup_cfg(cfg_file_path, parser: ArgumentParser) -> Optional[Dict[str, Any]]:
+    """Extract config mapping from setup.cfg file."""
+    import configparser
+
+    reader = configparser.ConfigParser()
+    reader.read(cfg_file_path)
+    if not reader.has_section('nbstripout'):
+        return None
+
+    raw_config = reader['nbstripout']
+    dict_config = dict(raw_config)
+
+    # special processing of boolean options, to convert various configparser bool types to true/false
+    for a in parser._actions:
+        if a.dest in raw_config and isinstance(a, (_StoreTrueAction, _StoreFalseAction)):
+            dict_config[a.dest] = raw_config.getboolean(a.dest)
+
+    return dict_config
+
+
+def merge_configuration_file(parser: ArgumentParser, args_str=None) -> Namespace:
+    """Merge flags from config files into args."""
+    CONFIG_FILES = {
+        'pyproject.toml': partial(process_pyproject_toml, parser=parser),
+        'setup.cfg': partial(process_setup_cfg, parser=parser),
+    }
+
+    # parse args as-is to look for configuration files
+    args = parser.parse_args(args_str)
+
+    # Traverse the file tree common to all files given as argument looking for
+    # a configuration file
+    # TODO: make this more like Black:
+    #       By default Black looks for pyproject.toml starting from the common base directory of all files and
+    #       directories passed on the command line. If it’s not there, it looks in parent directories. It stops looking
+    #       when it finds the file, or a .git directory, or a .hg directory, or the root of the file system, whichever
+    #       comes first.
+    # if no files are given, start from cwd
+    config_path = os.path.commonpath([os.path.abspath(file) for file in args.files]) if args.files else os.path.abspath(os.getcwd())
+    config: Optional[Dict[str, Any]] = None
+    while True:
+        for config_file, processor in CONFIG_FILES.items():
+            config_file_path = os.path.join(config_path, config_file)
+            if os.path.isfile(config_file_path):
+                config = processor(config_file_path)
+                if config is not None:
+                    break
+        if config is not None:
+            break
+        config_path, tail = os.path.split(config_path)
+        if not tail:
+            break
+
+    # black starts with default arguments (from click), updates that with the config file,
+    # then applies command line arguments. this all happens in the click framework, before main() is called
+    # we can use parser.set_defaults
+    if config:
+        # check all arguments are valid
+        valid_args = vars(args).keys()
+        for name in config.keys():
+            if name not in valid_args:
+                raise ValueError(f'{name} in the config file is not a valid option')
+
+        # separate into default-overrides and special treatment
```

Comment:
I'm not 100% sure that it does. I left this with the same functionality that @janosh did.
This special treatment lets you specify a base set of extra keys in the config files, and then add to that in an ad-hoc manner, but does not let you remove extra keys....
Alternative treatment would be that whenever you specify extra keys you would have to list the complete set.
The more I think about it, the more I think that perhaps special treatment is NOT justified.

Reply:
Yes, I think that's a fair point and treating it specially may lead to surprising results.

@janosh can you explain your rationale? Do you agree that special treatment of this flag may be surprising?

Reply:
I don't remember for sure at this point, but I think it might be left over from the original autoflake8 config file PR. I agree minimal user surprise is usually the way to go.